File: Exercism Codes/JavaScript/simple-cipher.js
export class Cipher {
  constructor(key) {
    if (key === undefined) {
      this._key = new Array(100).fill(0).map(_ => String.fromCharCode(Math.floor(Math.random() * 26) + 97)).join("");
    }
    else {
      this._key = key;
    }
  }

  encode(text = "") {
    return text.split("").map((x, idx) => {
      const res = (x.charCodeAt(0) - 97) + (this._key.charCodeAt(idx % this._key.length) - 97);
      const result = 97 + (res % 26);
      return String.fromCharCode(result);
    }).join("");
  }

  decode(text) {
    return text.split("").map((x, idx) => {
      const res = (x.charCodeAt(0) - 97) - (this._key.charCodeAt(idx % this._key.length) - 97);
      const result = 97 + ((res % 26) + (res < 0 ? 26 : 0));
      return String.fromCharCode(result);
    }).join("");
  }

  get key() {
    return this._key;
  }
}

File: Exercism Codes/Python/simple_cipher.py
import random
import string


class Cipher:
    def __init__(self, key=None):
        if key is None:
            key = "".join(random.choice(string.ascii_lowercase) for _ in range(100))

        self.key = key

    def encode(self, text):
        return "".join(chr(97 + ((ord(x) - 97) + (ord(self.key[idx % len(self.key)]) - 97)) % 26) for idx, x in enumerate(text))

    def decode(self, text):
        return "".join(chr(97 + ((ord(x) - 97) - (ord(self.key[idx % len(self.key)]) - 97)) % 26) for idx, x in enumerate(text))
